### PR TITLE
#158518839 Add user sign-up endpoint feature.

### PR DESCRIPTION
**Pull request by meshack-mbuvi**
#### What does this PR do?
Add endpoint for user sign-up
#### Description of Task to be completed?
- Add view function to for sign-up endpoint
- Implement actual user sign-up within the view function
```
- POST /auth/signup  Register a user   
```
#### How should this be manually tested?
using your terminal,
- clone this repo
- `cd ride-my-way`
- set and activate your virtual environment
- `git checkout ft-user-signup-158518839`
- Run `python3 run.py`
- Using postman, access 'your-server/api/v1/auth/signup`

**screenshots**
user-sign-up
user-signup

**Pivotal tracker stories **
[user sign up #158518839](https://www.pivotaltracker.com/story/show/158518839)

### CODE REVIEW

**Comment by TheSteelGuy** [comment]
seems good

File: application/manage.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                       owner_id serial, \
+                       destination varchar(255), \
+                       start_time varchar(50) NOT NULL, \
+                       route varchar(255) NOT NULL, \
+                       available_space Int NOT NULL, \
+                       FOREIGN KEY (owner_id) REFERENCES users(user_id) )'
+        )
+
+        try:
+            cur = con.cursor()
```
Comment (LWanjiru) [inline application/manage.py]: Please consider using self-describing variable names, remember other people might work with your codebase in future. How will they identify these variables?

Comment (meshack-mbuvi) [reply application/manage.py]: Ok, working on that.

File: application/__init__.py [hunk 2]
```diff
@@ -19,8 +27,20 @@ def create_app(config):
...
+    @jwt.token_in_blacklist_loader
+    def check_if_token_in_blacklist(decrypted_token):
+        jti = decrypted_token['jti']
+        return jti in blacklist
+
     from application.views.ride_views import api as rides
+    from application.views.user_views import api as user
     # Blueprints to be registered here
-
     api.add_namespace(rides, path='/api/v1')
```
[the remark this answers is not in the document]
Comment (meshack-mbuvi) [reply application/__init__.py]: I get your point

[the remark this answers is not in the document]
Comment (meshack-mbuvi) [reply application/__init__.py]: ok, its my v1 of challenge three. I will look at product versions and what it entails

**Comment by LWanjiru** [comment]
Fix conflicts... also your build is failing

**Comment by meshack-mbuvi** [comment]
Conflicts resolved.

File: application/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (xcixor) [reply application/__init__.py]: awesome

[the remark this answers is not in the document]
Comment (emkuchela) [reply application/__init__.py]: great work here

**Comment by meshack-mbuvi** [comment]
I have ESlint

**Comment by meshack-mbuvi** [comment]
@LWanjiru Does look professional now?

**Pull request merged**